exchange/wrappers/gdax.js

```diff
@@ -115,117 +112,148 @@ Trader.prototype.getPortfolio = function(callback) {
-  const fee = this.asset == 'BTC' ? 0.0025 : 0.003;
+  // const fee = this.asset == 'BTC' ? 0.0025 : 0.003;
+  const fee = 0;
 
   //There is no maker fee, not sure if we need taker fee here
   //If post only is enabled, gdax only does maker trades which are free
   callback(undefined, this.post_only ? 0 : fee);
 };
 
+Trader.prototype.roundPrice = function(price) {
+  return this.getMaxDecimalsNumber(price, this.currency == 'BTC' ? 5 : 2);
+}
+
+Trader.prototype.roundAmount = function(amount) {
+  return this.getMaxDecimalsNumber(amount);
+}
+
 Trader.prototype.buy = function(amount, price, callback) {
-  var buyParams = {
+  const buyParams = {
     price: this.getMaxDecimalsNumber(price, this.currency == 'BTC' ? 5 : 2),
     size: this.getMaxDecimalsNumber(amount),
     product_id: this.pair,
     post_only: this.post_only,
   };
 
-  var result = (err, data) => {
-    if (err) return callback(err);
+  const result = (err, data) => {
+    if (err) {
+      console.log({buyParams}, err.message);
+      return callback(err);
+    }
     callback(undefined, data.id);
   };
 
-  let handler = cb =>
-    this.gdax.buy(buyParams, this.handleResponse('buy', cb));
-  util.retryCustom(retryCritical, _.bind(handler, this), _.bind(result, this));
+  const fetch = cb =>
+    this.gdax.buy(buyParams, this.processResponse('buy', cb));
+  retry(null, fetch, result);
 };
 
 Trader.prototype.sell = function(amount, price, callback) {
-  var sellParams = {
+  const sellParams = {
     price: this.getMaxDecimalsNumber(price, this.currency == 'BTC' ? 5 : 2),
     size: this.getMaxDecimalsNumber(amount),
     product_id: this.pair,
     post_only: this.post_only,
   };
 
-  var result = function(err, data) {
-    if (err) return callback(err);
+  const result = (err, data) => {
+    if (err) {
+      console.log({sellParams}, err.message);
+      return callback(err);
+    }
+
+    if(data.message && data.message.includes('Insufficient funds')) {
+      err = new Error(data.message);
+      err.retryOnce = true;
+      return callback(err);
+    }
+
     callback(undefined, data.id);
   };
 
-  let handler = cb =>
-    this.gdax.sell(sellParams, this.handleResponse('sell', cb));
-  util.retryCustom(retryCritical, _.bind(handler, this), _.bind(result, this));
+  const fetch = cb =>
+    this.gdax.sell(sellParams, this.processResponse('sell', cb));
+  retry(null, fetch, result);
 };
 
 Trader.prototype.checkOrder = function(order, callback) {
-  var result = function(err, data) {
+  const result = (err, data) => {
     if (err) return callback(err);
 
+    // @link:
+    // https://stackoverflow.com/questions/48132078/available-gdax-order-statuses-and-meanings
     var status = data.status;
-    if (status == 'done') {
-      return callback(undefined, true);
-    } else if (status == 'rejected') {
-      return callback(undefined, false);
-    } else if (status == 'pending') {
-      return callback(undefined, false);
+    if(status == 'pending') {
```

Comment:
could use a switch statement

Reply:
not a fan of switch statements for these kind of situations. I think this is more readable :(